pennylane/devices/default_qubit.py

```diff
@@ -171,47 +171,45 @@ def apply(self, operations, rotations=None, **kwargs):
                     "on a {} device.".format(operation.name, self.short_name)
                 )
 
-            self._apply_operation(operation)
+            if isinstance(operation, QubitStateVector):
+                self._apply_state_vector(operation.parameters[0], operation.wires)
+            elif isinstance(operation, BasisState):
+                self._apply_basis_state(operation.parameters[0], operation.wires)
+            else:
+                self._state = self._apply_operation(self._state, operation)
 
         # store the pre-rotated state
         self._pre_rotated_state = self._state
 
         # apply the circuit rotations
         for operation in rotations:
-            self._apply_operation(operation)
+            self._state = self._apply_operation(self._state, operation)
 
-    def _apply_operation(self, operation):
-        """Applies operations to the internal device state.
+    def _apply_operation(self, state, operation):
```

Comment:
I'm mostly going on feeling here, but what about:

```python
_apply_operation(operation, state)
```

My brain reads that more as "Apply `operation` onto `state`" instead of "update `state` with `operation`".  It just seems a bit easier to process and remember.

Reply:
Nice that could work! In the interests of time I'd rather stick with the current, but we could consider swapping around post-release.